ecosystem/typescript/sdk/src/aptos_client.test.ts

```diff
@@ -227,3 +228,118 @@ test(
+    await tokenClient.createCollection(alice, collectionName, "Alice's simple collection", 'https://aptos.dev');
+
+    await tokenClient.createToken(
+      alice,
+      collectionName,
+      tokenName,
+      // eslint-disable-next-line quotes
+      "Alice's simple token",
+      1,
+      'https://aptos.dev/img/nyan.jpeg',
+      0,
+    );
+
+    let aliceBalance = await tokenClient.getTokenBalance(alice.address().hex(), collectionName, tokenName);
+    expect(aliceBalance.value).toBe('1');
+
+    const scriptFunctionPayload = new TxnBuilderTypes.TransactionPayloadScriptFunction(
+      TxnBuilderTypes.ScriptFunction.natual(
+        '0x1::Token',
+        'direct_transfer_script',
+        [],
+        [
+          BCS.bcsToBytes(aliceAccountAddress),
+          BCS.bcsSerializeStr(collectionName),
+          BCS.bcsSerializeStr(tokenName),
+          BCS.bcsSerializeUint64(1),
+        ],
+      ),
+    );
+
+    const [{ sequence_number: sequnceNumber }, chainId] = await Promise.all([
+      client.getAccount(alice.address()),
+      client.getChainId(),
+    ]);
+
+    const rawTxn = new TxnBuilderTypes.RawTransaction(
+      aliceAccountAddress,
+      BigInt(sequnceNumber),
+      scriptFunctionPayload,
+      1000n,
+      1n,
+      BigInt(Math.floor(Date.now() / 1000) + 10),
+      new TxnBuilderTypes.ChainId(chainId),
+    );
+
+    const multiAgentTxn = new TxnBuilderTypes.MultiAgentRawTransaction(rawTxn, [bobAccountAddress]);
+
+    const aliceSignature = new TxnBuilderTypes.Ed25519Signature(
+      alice.signBuffer(TransactionBuilder.getSigningMessage(multiAgentTxn)).toUint8Array(),
+    );
+
+    const aliceAuthenticator = new TxnBuilderTypes.AccountAuthenticatorEd25519(
+      new TxnBuilderTypes.Ed25519PublicKey(alice.signingKey.publicKey),
+      aliceSignature,
+    );
+
+    const bobSignature = new TxnBuilderTypes.Ed25519Signature(
+      bob.signBuffer(TransactionBuilder.getSigningMessage(multiAgentTxn)).toUint8Array(),
+    );
+
+    const bobAuthenticator = new TxnBuilderTypes.AccountAuthenticatorEd25519(
+      new TxnBuilderTypes.Ed25519PublicKey(bob.signingKey.publicKey),
+      bobSignature,
+    );
+
+    const multiAgentAuthenticator = new TxnBuilderTypes.TransactionAuthenticatorMultiAgent(
+      aliceAuthenticator, // sender authenticator
+      [bobAccountAddress], // secondary signer addresses
+      [bobAuthenticator], // secondary signer authenticators
+    );
+
+    const bcsTxn = BCS.bcsToBytes(new TxnBuilderTypes.SignedTransaction(rawTxn, multiAgentAuthenticator));
+
+    const transactionRes = await client.submitSignedBCSTransaction(bcsTxn);
+
+    await client.waitForTransaction(transactionRes.hash);
+
+    const transaction = await client.transactions.getTransaction(transactionRes.hash);
+    expect((transaction.data as any)?.success).toBe(true);
+
+    aliceBalance = await tokenClient.getTokenBalance(alice.address().hex(), collectionName, tokenName);
+
+    expect(aliceBalance.value).toBe('0');
```

Comment:
why not check Bob's?

Reply:
because there are no on-chain interfaces to query the receiver token balance. Have to leverage the indexer.

Reply:
This is the same as the simple token transfer. The token now lives in the recipients tokenstore. This is definitely queryable.